### PR TITLE
Instructions for allow/deny list

### PR DESCRIPTION
**Pull request by Anxhela21**
Signed-off-by: Anxhela <acoba@redhat.com>

**Related Issue:**  open-cluster-management/backlog#20225 for Dev Preview - Search Configurable Collections

### Description of changes
Dev writeup for how to use control resources.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
 - Environment variables can also be set in the `./config.json` for development. If both provide a value for a specific property, the environment variable overrides the file. You can define your own `config.json` file and pass it to the application with the following command: `-c <config_file>`
 - The application can take any flags for [glog](https://github.com/golang/glog), which passes them straight into glog. The glog flag `--logtostderr` is set to true by default.
 
+### Dev Preview (Search Configurable Collection)
+
+Control resources that get collected from the Search-Collector by referencing an allow and deny list within search-collector-config configmap. Follow the following format in the sample template below:
```
Comment (jlpadilla) [inline readme.md:61]:
```suggestion
Control the Kubernetes resources that get collected from the cluster by referencing an allow and deny list within a configmap with the name search-collector-config. Create the configmap following format in the sample template below:
```

File: readme.md [hunk 2]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+
+
+<pre><code>
+apiVersion: v1
+kind: ConfigMap
+metadata:
+ name: search-collector-config
+ namespace: Namespace where Search-Collector Pod is
```
Comment (jlpadilla) [inline readme.md:69]: Let's use `< >` to indicate that the value needs to be replaced. 
```suggestion
 namespace: <Namespace where search collector add-on is deployed.>
```

Here's an example [from our product docs.](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/install/installing#installing-from-the-cli)

File: readme.md [hunk 3]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+       - policies
+       - iampolicies
+       - certificatepolicies
+</code></pre>
+
+Steps to create search-collector-config
+
+1. The <b>Namespace</b> value should be defined as the namespace where the Search-Collector pod lives.
+
+2. Use the <b>Name</b> `search-collector-config`
```
Comment (jlpadilla) [inline readme.md:99]: - Move **Name** first, to match order of sample.
- Indicate that using this name is required.
- `<b>bold</b>` works, but I prefer use markdown notation `**bold**` consistently.

File: readme.md [hunk 4]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+       - admission.k8s.io
+     resources:
+       - policies
+       - iampolicies
+       - certificatepolicies
+</code></pre>
+
+Steps to create search-collector-config
+
+1. The <b>Namespace</b> value should be defined as the namespace where the Search-Collector pod lives.
```
Comment (jlpadilla) [inline readme.md:97]:
```suggestion
1. **name** of the ConfigMap must be `search-collector-config`.
2. **namespace** is the Namespace where the search collector add-on is deployed.
```

File: readme.md [hunk 5]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+
+2. Use the <b>Name</b> `search-collector-config`
+
+3. Under <b>data</b> define `AllowedResources` and `DeniedResources` as key value pairs wrapped in a string block with `|-` to preserve linebreaks.
+
+    - The asterisk `"*"` represents <i>all</i>.
+
+    - Note that if you want to control resources for the current group then you should replace the `apiGroups` value with and empty string `""`.
```
Comment (jlpadilla) [inline readme.md:105]: Remove this statement, it's not accurate.

FYI, some resources don't have an apigroup, for those the apigroup is represented with an empty string. The resources that have an empty apigroup are the base kubernetes resources (Pod,ConfigMap,Secret, etc). All other extensions to the basic resources will have an apigroup.

Here's how to find the resources without a group. The resources at the top of this list only have version (v1) and no apigroup.
image

File: readme.md [hunk 6]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+    - Note that if you want to control resources for the current group then you should replace the `apiGroups` value with and empty string `""`. 
+    - If the same resources are featured in both lists, they will be excluded.
+4. Once you save your changes you can apply your changes by running `oc apply -f configMapFile.yaml`
+5. Add the following env variable to your Search-Collector deployment:
+
+    <pre><code> - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace</code></pre>
```
Comment (jlpadilla) [inline readme.md:114]: This step isn't needed. We updated the chart to always include this.

File: readme.md (hunk 5 above)
Comment (Anxhela21) [reply readme.md:105]: Ah, okay that makes sense. Thanks for clarifying!

File: readme.md [hunk 7]
```diff
@@ -56,6 +56,67 @@ RUNTIME_MODE       | no       | production               | Running mode (develop
...
+6. Restart the Search-Collector pod
```
[the remark this answers is not in the document]
Comment (Anxhela21) [reply readme.md:115]: Okay, we have been testing by restarting the collector (that is also what is mentioned in the [user story](https://github.com/stolostron/backlog/issues/20225)) but I can remove this step if it works without it.

[the remark this answers is not in the document]
Comment (Anxhela21) [reply readme.md:115]: or maybe leave as optional step?

File: readme.md (hunk 6 above)
Comment (Anxhela21) [reply readme.md:114]: I'll need to check again but I recall having to update this manually. Will remove the step.

File: readme.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jlpadilla) [reply readme.md]: Sorry, let's keep the restart step.

I tested it and the new code picks up the changes but there's a bug in another part that doesn't update the state then a new resource gets denied.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**